o4/o4.py

```diff
@@ -827,29 +854,76 @@ def man(cmd):
+
+    # [{
+    #     'code': 'stat',
+    #     'depotFile': '//sandbox/t/a',
+    #     'clientFile': '//pbergen-ltm-blt/sandbox/t/a',
+    #     'movedFile': '//sandbox/t/b',
+    #     'rev': '1',
+    #     'haveRev': '2',
+    #     'action': 'move/delete',
+    #     'change': 'default',
+    #     'type': 'text',
+    #     'user': 'pbergen',
+    #     'client': 'pbergen-ltm-blt'
+    # }, {
+    #     'code': 'stat',
+    #     'depotFile': '//sandbox/t/b',
+    #     'clientFile': '//pbergen-ltm-blt/sandbox/t/b',
+    #     'movedFile': '//sandbox/t/a',
+    #     'rev': '2',
+    #     'haveRev': '2',
+    #     'action': 'move/add',
+    #     'change': 'default',
+    #     'type': 'text',
+    #     'user': 'pbergen',
+    #     'client': 'pbergen-ltm-blt'
+    # }]
+
     openf = NamedTemporaryFile(dir='.o4', mode='w+t')
     if has_open:
+        from shlex import quote as shquote
+        move_adds = []
+        move_deletes = []
+
         dep = _depot_path().replace('/...', '')
         print(f'*** INFO: Opened for edit in {dep}:')
         for i, p in enumerate(has_open):
             open_file_name = Pyforce.unescape(p['depotFile'])[len(dep) + 1:]
             print(open_file_name, file=openf)
+            if p['action'] == 'move/add':
+                move_adds.append(shquote(open_file_name))
+            if p['action'] == 'move/delete':
+                move_deletes.append(shquote(open_file_name))
             if i < 10:
                 print(f'*** INFO: --keeping {open_file_name}')
         if len(has_open) > 10:
             print(f'          (and {len(has_open) - 10} more)')
         openf.flush()
 
         # Resolve before syncing in case there are unresolved files for other reasons
-        cmd = (f"{fstat} --keep {openf.name}"
-               f"| {gatling_verbose} -- {o4bin} {pyforce} --no-rev -- resolve -am"
-               f"| {o4bin} {pyforce} sync"
-               f"| {gatling_verbose} -- {o4bin} {pyforce} --no-rev -- resolve -am"
-               f"{progress}"
-               f"| {o4bin} drop --existence"
-               f"| {gatling_verbose} -- {o4bin} {pyforce} --no-rev -- revert"
-               f"| {o4bin} drop --existence"
-               f"| {o4bin} fail")
+        move_add = ''
+        if move_adds:
+            move_add = ' --add '.join([''] + move_adds)
+        move_delete = ''
+        if move_deletes:
+            move_delete = ' --deleted '.join([''] + move_deletes)
+            move_delete = f"| {o4bin} drop {move_delete}"
+
+        cmd = (
+            f"{fstat} --keep {openf.name}{move_add}"
+            f"| {gatling_verbose} -- {o4bin} {pyforce} --no-rev -- resolve -am"
+            f"| {o4bin} {pyforce} sync"
+            f"| {gatling_verbose} -- {o4bin} {pyforce} --no-rev -- resolve -am"
+            f"{progress}"
+            f"{move_delete}"
+            f"| {o4bin} drop --existence"
+            # Why this revert? Doesn't that risk losing all local changes??
```

Comment:
The local change lost would be in a file that's not supposed to be there anyway, right? (Because of the previous line.)